configure-android.sh

```diff
@@ -1,199 +1,223 @@
+            shift 2
+            ;;
+        --api)
+            is_integer "$2" || die "--api API, API must be a integer."
+            if [ "$2" -ge 21 ] && [ "$2" -le 30 ] ; then
+                ANDROID_API_LEVEL=$2
+                shift 2
+            else
+                die "--api API, API must be between 21 and 30."
+            fi
+            ;;
+        --libpcap-include-dir)
+            LIBPCAP_INCLUDE_DIR=$2
+            shift 2
+            ;;
+        --libpcap-lib-dir)
+            LIBPCAP_LIBRARY_DIR=$2
+            shift 2
+            ;;
+        *)  die "unrecognized option: $1"
     esac
 done
 
-if [ -z "$NDK_PATH" ]; then
-    echo "Please specify the NDK path using with '--ndk-path'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_NDK_ROOT" ] ; then
+    die "please specify the Android NDK root path via --ndk-path PATH"
 fi
 
-if [ -z "$TARGET" ]; then
-    echo "Please specify the target using wity '--target'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_API_LEVEL" ] ; then
+    ANDROID_API_LEVEL=29
+fi
+
+if [ -z "$TARGET_TRIPLE" ] ; then
+    die "please specify the target via --target TARGET"
 fi
 
 if [ -z "$LIBPCAP_INCLUDE_DIR" ]; then
-    echo "Please specify the location of libpcap header files with '--libpcap-include-dir'. Exiting..."
-    exit 1
+    die "please specify the directory of libpcap header files via --libpcap-include-dir DIR"
 fi
 
-if [ -z "$LIBPCAP_LIB_DIR" ]; then
-    echo "Please specify the location of libpcap binary that matches the reqruied arch with '--libpcap-lib-dir'. Exiting..."
-    exit 1
+if [ -z "$LIBPCAP_LIBRARY_DIR" ]; then
+    die "please specify the directory of libpcap binary that matches the specified target via --libpcap-lib-dir DIR"
 fi
 
-PLATFORM_MK="mk/platform.mk"
-PCAPPLUSPLUS_MK="mk/PcapPlusPlus.mk"
+#########################################################################################
+
+unset BUILD_MACHINE_OS_TYPE
+unset BUILD_MACHINE_OS_ARCH
+
+BUILD_MACHINE_OS_TYPE=$(uname | tr A-Z a-z)
+BUILD_MACHINE_OS_ARCH=$(uname -m)
+
+ANDROID_TOOLCHAIN=$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/$BUILD_MACHINE_OS_TYPE-$BUILD_MACHINE_OS_ARCH
+
+#########################################################################################
+
+cat <<EOF
+BUILD_MACHINE_OS_TYPE=$BUILD_MACHINE_OS_TYPE
+BUILD_MACHINE_OS_ARCH=$BUILD_MACHINE_OS_ARCH
+
+ANDROID_NDK_ROOT =$ANDROID_NDK_ROOT
+ANDROID_API_LEVEL=$ANDROID_API_LEVEL
+ANDROID_TOOLCHAIN=$ANDROID_TOOLCHAIN
+
+TARGET_TRIPLE=$TARGET_TRIPLE
+
+LIBPCAP_INCLUDE_DIR=$LIBPCAP_INCLUDE_DIR
+LIBPCAP_LIBRARY_DIR=$LIBPCAP_LIBRARY_DIR
+EOF
+
+#########################################################################################
+
+unset TEMP_DIR
+unset TEMP_CONFIG_FILE
+
+TEMP_DIR=$(mktemp -d)
+TEMP_CONFIG_FILE="$TEMP_DIR/config.mk"
+
+cat > "$TEMP_CONFIG_FILE" <<EOF
+PCAPPLUSPLUS_HOME   = $PWD
+ANDROID_TARGET      = $TARGET_TRIPLE$ANDROID_API_LEVEL
+ANDROID_NDK_PATH    = $ANDROID_NDK_ROOT
+ANDROID_API_VERSION = $ANDROID_API_LEVEL
```

Comment:
I think it's safer to use `:=` (simply expanded variable) than `=` (recursively expanded variable)